### PR TITLE
Added minor/major transpose to 4th order tensors

### PR DESCRIPTION
**Pull request by fredrikekre**
... and functions for making them major/minor symmetric

added `Compat` also, not sure if it is correct, but test passes on 0.4 and 0.5 atleast :)

on 0.4 I get this warning at start of `runtest.jl`

``` jl
julia> include("test/runtests.jl")
WARNING: imported binding for Test overwritten in module Main
```

not sure where it is overwritten and why.

### CODE REVIEW

File: src/promotion_conversion.jl [hunk 1]
```diff
@@ -165,7 +165,7 @@ Base.convert{order, dim, T, M}(::Type{SymmetricTensor}, t::Tensor{order, dim, T,
 
 
 # Tensor -> SymmetricTensor
-function Base.issymmetric{dim}(t::Tensor{2, dim})
+@compat function issymmetric{dim}(t::Tensor{2, dim})
```
Comment (KristofferC) [inline src/promotion_conversion.jl]: These `@compat` shouldn't be needed I think.

File: src/tensor_ops.jl [hunk 2]
```diff
@@ -241,6 +241,17 @@ Computes the deviatoric part of a second order tensor.
     end
 end
 
+###################
+# Permute indices #
+###################
+
+function permute_index{dim}(S::FourthOrderTensor{dim},idx::NTuple{4,Int})
```
Comment (KristofferC) [inline src/tensor_ops.jl:248]: This function will be quite inefficient but no problems.

File: test/test_ops.jl [hunk 3]
```diff
@@ -176,6 +197,43 @@ for dim in (1,2,3)
...
+    @test majortranspose(majortranspose(AA)) ≈ AA
+    @test majortranspose(majortranspose(AA_sym)) ≈ AA_sym
+    @test AA[i,j,k,l] ≈ majortranspose(AA)[k,l,i,j]
+    @test AA_sym[i,j,k,l] ≈ majortranspose(AA_sym)[k,l,i,j]
+    @test typeof(majortranspose(AA_sym)) <: Tensor{4,dim}
+
+    #################
+    # Permute index #
+    #################
+    @test permute_index(AA,(1,2,3,4)) ≈ AA
```
Comment (KristofferC) [inline test/test_ops.jl:225]: Nice tests!

File: src/tensor_ops.jl (hunk 2 above)
Comment (fredrikekre) [inline src/tensor_ops.jl:248]: Yea I know,
Initially wrote this to use it with minor/major transpose. But realized the specialized edition which exist now are much better, decided to keep this anyway.

**Comment by KristofferC** [comment]
Looks good to me. You should be able to merge yourself now :)

**Comment by fredrikekre** [comment]
Yey!

**Pull request merged**